lib/snippets.coffee

```diff
@@ -273,9 +273,18 @@ module.exports =
   addExpansion: (editor, snippetExpansion) ->
     @getExpansions(editor).push(snippetExpansion)
 
-  insert: (snippet, editor=atom.workspace.getActiveTextEditor(), cursor=editor.getLastCursor()) ->
+  insert: (snippet, editor = atom.workspace.getActiveTextEditor(), cursor = editor.getLastCursor()) ->
     if typeof snippet is 'string'
       bodyTree = @getBodyParser().parse(snippet)
       snippet = new Snippet({name: '__anonymous', prefix: '', bodyTree, bodyText: snippet})
 
     new SnippetExpansion(snippet, editor, cursor, this)
+
+  provideSnippets: ->
+    isLoaded = -> @loaded
+    snippets = this
+    return s =
+      loaded: isLoaded.bind(snippets)
+      insert: @insert.bind(snippets)
+      getSnippets: @getSnippets.bind(snippets)
+      getPrefixText: @getPrefixText.bind(snippets)
```

Comment:
getPrefixText would be used to replace the default prefix that is provided by autocomplete-plus in the suggestion request: https://github.com/atom-community/autocomplete-snippets/blob/7e183365030cfa96586077d0e7eb339d5b52eace/lib/snippets-provider.coffee#L9. I would assume the snippets package has better logic for determining the prefix of a snippet than we do, but perhaps this is moot given the simplicity of most prefixes (they are generally `[\w]*`).